src/contexts/cart.context.js

```diff
@@ -0,0 +1,133 @@
+import axios from "axios";
+import { createContext, useContext, useReducer, useEffect } from "react";
+import { useAuth } from "./auth.context";
+import { cartReducer } from '../reducers'
+
+const CartContext = createContext()
+
+export const CartProvider = ({ children }) => {
+    const [cartState, cartDispatch] = useReducer(cartReducer, {
+        cart: [],
+        alert: {
+            message: '',
+            type: ''
+        },
+        loading: false
+    })
+    const { getUserToken, isUserLoggedIn } = useAuth()
+
+    useEffect(() => {
+        !isUserLoggedIn && cartDispatch({ type: 'INIT_CART', payload: [] })
+    }, [isUserLoggedIn])
+
+    function showCartAlert(message, type) {
+        cartDispatch({
+            type: 'SET_ALERT', payload: {
+                message,
+                type
+            }
+        })
+        setTimeout(() => {
+            cartDispatch({
+                type: 'REMOVE_ALERT', payload: {
+                    message: '',
+                    type: '',
+                }
+            })
+        }, 3000)
+    }
+
+    const isProductInCart = productId => cartState.cart.some(item => item._id === productId)
+
+    async function decreaseProductQuantity(qty, _id) {
+        if (qty > 0) {
+            await axios.post(`/api/user/cart/${_id}`, {
+                action: {
+                    type: 'decrement'
+                }
+            }, {
+                headers: {
+                    authorization: getUserToken()
+                }
+            })
+            cartDispatch({ type: 'DECREMENT_CART_PRODUCT', payload: _id })
+        }
+    }
+
+    async function increaseProductQuantity(_id) {
```

Comment:
You can merge increase and decrease product quantity by just taking one more argument ("actionType") in the function . 

`decreaseProductQuantity(qty, _id, actionType)`

And pass that type here

`action: { type: actionType}`

Reply:
will make the change.